### ISSUE TITLE
Incorrect mask indexing in token dropping implementation

### ISSUE THREAD

**Issue by v4ndi** [ahans30/goldfish-loss#3]
## Description
In the token dropping implementation, dropped_token_indices is created as an integer tensor but used directly for boolean indexing. This causes incorrect masking behavior in the "seeded_random" strategy. When applying .int() conversion with `k=2`, the resulting mask tensor contains only 0s, 1s. This means that only tokens at indices 0, 1 can be masked.

  
**Problem location:**
The issue occurs in the `seeded_random` branch where:
```python
    elif strategy == "seeded_random":
        random_tensor = torch.randint(1, k + 1, size=targets.size())
        dropped_token_indices = (random_tensor == k).int() # probability of dropping a token is 1/k
        masked_targets[dropped_token_indices] = ignore_index
```
**Expected Behavior**:  
For boolean indexing, the mask should be a boolean tensor:  
```python
drop_mask = (random_tensor == k)  # Creates bool tensor
masked_targets[drop_mask] = ignore_index
```

**Example:**
```python
import torch

# Example setup
k = 2
targets = torch.tensor([1, 2, 3, 5, 6]).long()  # Sample token IDs

# Generate random drop decisions (1/k probability)
random_tensor = torch.randint(1, k + 1, size=targets.size())

# INCORRECT: Using .int() creates [0,1] indices (wrong for masking)
incorrect_mask = (random_tensor == k).int()  

# CORRECT: Boolean mask (what we actually want)
correct_mask = (random_tensor == k)  

print("Random decisions:", random_tensor)
print("Incorrect mask (as ints):", incorrect_mask)
print("Correct mask (as bools):", correct_mask)

# Apply both masking approaches
incorrect_masked = targets.clone()
incorrect_masked[incorrect_mask] = -100  # WRONG: Uses integer indexing

correct_masked = targets.clone()
correct_masked[correct_mask] = -100  # CORRECT: Uses boolean masking

print("\nResults:")
print("Original targets:    ", targets.tolist())
print("Incorrect masking:   ", incorrect_masked.tolist())
print("Correct masking:     ", correct_masked.tolist())
```
